client/src/app.tsx

```diff
@@ -16,21 +16,27 @@ function App() {
   const appId = process.env.HATHORA_APP_ID;
   const token = useAuthToken(appId);
   const [connection, setConnection] = useState<HathoraConnection | undefined>();
-  const [displayMetadata, setDisplayMetadata] = useState<DisplayMetadata>({ serverUrl: "" });
+  const [displayMetadata, setDisplayMetadata] = useState<DisplayMetadata>({ serverUrl: "", winningScore: 15 });
   const [failedToConnect, setFailedToConnect] = useState(false);
 
   const joinRoom = useCallback(
-    (lobbyClient: PlayerLobbyClient<LobbyState>) => (roomId: string) =>
+    (lobbyClient: PlayerLobbyClient<LobbyState, InitialConfig>) => (roomId: string) =>
       lobbyClient
         .getConnectionDetailsForLobbyV2(roomId, { host: "localhost", port: 4000, transportType: "tcp" as const })
-        .then((connectionDetails) => {
+        .then(async (connectionDetails) => {
           if (connection != null) {
             connection.disconnect(200);
           }
+
+          const res = await lobbyClient.getLobbyInfoV2(roomId);
```

Comment:
it seems wrong to me to call getLobbyInfoV2 here given that initialConfig.winningScore is actually returned in createXLobby which is what was called right before joinRoom. Why don't you pass the entire Lobby object in as the argument to joinRoom here instead of the roomId so you don't have to make a second API call

Reply:
That works for the `Create Lobby` path, but I don't think we have it for free for the other join flows (`Lobby List Join` and `Room ID Join`

Reply:
oh yeah you're right